src/main/java/org/logstash/plugins/inputs/http/NettyHttpServer.java

```diff
@@ -0,0 +1,74 @@
+package org.logstash.plugins.inputs.http;
+
+import io.netty.bootstrap.ServerBootstrap;
+import io.netty.channel.*;
+import io.netty.channel.nio.NioEventLoopGroup;
+import io.netty.channel.socket.nio.NioServerSocketChannel;
+import io.netty.util.concurrent.DefaultEventExecutorGroup;
+import org.logstash.plugins.inputs.http.util.SslSimpleBuilder;
+
+import java.io.Closeable;
+import java.util.concurrent.TimeUnit;
+
+import static org.logstash.plugins.inputs.http.util.DaemonThreadFactory.daemonThreadFactory;
+
+/**
+ * Created by joaoduarte on 11/10/2017.
+ */
+public class NettyHttpServer implements Runnable, Closeable {
+    private final int numProcessors = Runtime.getRuntime().availableProcessors();
+    private final ServerBootstrap serverBootstrap;
+    private final String host;
+    private final int port;
+    private SslSimpleBuilder sslSimpleBuilder;
+
+    private final EventLoopGroup processorGroup =
+            new NioEventLoopGroup(numProcessors, daemonThreadFactory("http-input-processor"));
+
+    private final DefaultEventExecutorGroup executorGroup = new DefaultEventExecutorGroup(4,
```

Comment:
We probably want to expose the thread and pending tasks as configuration options, and may want to default to some function of the number of cores.